staging/src/k8s.io/kubectl/pkg/cmd/apply/applyset.go

```diff
@@ -0,0 +1,114 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package apply
+
+import (
+	"fmt"
+	"strings"
+
+	"k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+const defaultApplySetParentResource = "secrets"
+const defaultApplySetParentVersion = "v1"
+
+// ApplySet tracks the information about an applyset apply/prune
+type ApplySet struct {
+	// ParentRef is the reference to the parent object that is used to track the applyset.
+	ParentRef *ApplySetParentRef
+
+	// resources is the set of all the resources that (might) be part of this applyset.
+	resources map[schema.GroupVersionResource]struct{}
+
+	// namespaces is the set of all namespaces that (might) contain objects that are part of this applyset.
+	namespaces map[string]struct{}
+}
+
+// ApplySetParentRef stores object and type meta for the parent object that is used to track the applyset.
+type ApplySetParentRef struct {
+	Name        string
+	Namespace   string
+	RESTMapping *meta.RESTMapping
+}
+
+const invalidParentRefFmt = "invalid parent reference %q: %w"
```

Comment:
My 2c is that extracting this to a constant obscures more than it helps, but I know others disagree.

Reply:
I don't feel strongly either way tbh